### ISSUE TITLE
Not able to navigate, after saving user profile

### ISSUE THREAD

**Issue by jdar** [couchbase-examples/flutter_cbl_learning_path#4]
SUMMARY:
Not able to navigate (except via the Menu), after saving user profile once. The pages are navigable again after a restart.

VERSION: 
I have only tested on Android 12. 
latest release of cbl (2.1.3) ... the bug was observed after testing the fix for #1 

STEPS TO REPLICATE:
Loaded in debug mode, from VS Code
On android, sign in as 'demo@example.com'
Open the menu and click 'Update User Profile'
Enter values ( entered in all 3 fields). Click 'Save'
A red Error screen flashes briefly. See logs below  (FIRST LOGS)
Then, attempt to click on a Project or on the '+' or on 'Update User Profile' 

EXPECTED:
The button depresses, and then the expected page loads.

ACTUAL:
The buttons depress, but no effect occurs 
See new error message in the logs, on each button press (SECOND LOGS)

ADDITIONAL THINGS:
After restarting the app (Hot Restart) the pages are navigable again, and it is possible to see that Update User Profile does show the updated data.

FIRST LOGS

```
════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown building BlocBuilder<UserProfileEditorBloc, UserProfileEditorState>(dirty, dependencies: [_InheritedProviderScope<UserProfileEditorBloc?>], state: _BlocBuilderBaseState<UserProfileEditorBloc, UserProfileEditorState>#7f178):
setState() or markNeedsBuild() called during build.

This Scaffold widget cannot be marked as needing to build because the framework is already in the process of building widgets. A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was: Scaffold
    dirty
    dependencies: [Directionality, MediaQuery, UnmanagedRestorationScope, _InheritedTheme, _LocalizationsScope-[GlobalKey#d7a1d], _ScaffoldMessengerScope]
    state: ScaffoldState#aecea(tickers: tracking 2 tickers)
The widget which was currently being built when the offending call was made was: BlocBuilder<UserProfileEditorBloc, UserProfileEditorState>
    dirty
    dependencies: [_InheritedProviderScope<UserProfileEditorBloc?>]
    state: _BlocBuilderBaseState<UserProfileEditorBloc, UserProfileEditorState>#7f178
The relevant error-causing widget was
BlocBuilder<UserProfileEditorBloc, UserProfileEditorState>
lib/…/views/user_profile_editor_widget.dart:128
When the exception was thrown, this was the stack

#0      Element.markNeedsBuild.<anonymous closure>
package:flutter/…/widgets/framework.dart:4549
#1      Element.markNeedsBuild
package:flutter/…/widgets/framework.dart:4564
#2      State.setState
package:flutter/…/widgets/framework.dart:1134
#3      ScaffoldState._drawerOpenedCallback
package:flutter/…/material/scaffold.dart:2033
#4      DrawerControllerState.close
package:flutter/…/material/drawer.dart:530
#5      DrawerControllerState._handleHistoryEntryRemoved
package:flutter/…/material/drawer.dart:431
#6      LocalHistoryEntry._notifyRemoved
package:flutter/…/widgets/routes.dart:484
#7      LocalHistoryRoute.didPop
package:flutter/…/widgets/routes.dart:689
#8      _RouteEntry.handlePop
package:flutter/…/widgets/navigator.dart:2900
#9      NavigatorState._flushHistoryUpdates
package:flutter/…/widgets/navigator.dart:3892
#10     NavigatorState.pop
package:flutter/…/widgets/navigator.dart:4967
#11     _SaveButton.build.<anonymous closure>
lib/…/views/user_profile_editor_widget.dart:131
#6      LocalHistoryEntry._notifyRemoved
package:flutter/…/widgets/routes.dart:484
#7      LocalHistoryRoute.didPop
package:flutter/…/widgets/routes.dart:689
#8      _RouteEntry.handlePop
package:flutter/…/widgets/navigator.dart:2900
#9      NavigatorState._flushHistoryUpdates
package:flutter/…/widgets/navigator.dart:3892
#10     NavigatorState.pop
package:flutter/…/widgets/navigator.dart:4967
#11     _SaveButton.build.<anonymous closure>
lib/…/views/user_profile_editor_widget.dart:131
#12     BlocBuilder.build
package:flutter_bloc/src/bloc_builder.dart:90
#13     _BlocBuilderBaseState.build
package:flutter_bloc/src/bloc_builder.dart:166
#14     StatefulElement.build
package:flutter/…/widgets/framework.dart:4992
#15     ComponentElement.performRebuild
.... [additional lines omitted]
```

SECOND LOGS

```
E/flutter (12668): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/widgets/navigator.dart': Failed assertion: line 4055 pos 12: '!_debugLocked': is not true.
package:flutter/…/widgets/navigator.dart:4055
E/flutter (12668): #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
E/flutter (12668): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
E/flutter (12668): #2      NavigatorState._routeNamed
package:flutter/…/widgets/navigator.dart:4055
E/flutter (12668): #3      NavigatorState.pushNamed
E/flutter (12668): #4      _AppViewState.build.<anonymous closure>.<anonymous closure>
package:flutter_cbl_learning_path/app_view.dart:75
E/flutter (12668): #5      _BlocListenerBaseState._subscribe.<anonymous closure>
package:flutter_bloc/src/bloc_listener.dart:204
E/flutter (12668): #6      _RootZone.runUnaryGuarded (dart:async/zone.dart:1586:10)
E/flutter (12668): #7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
E/flutter (12668): #8      _DelayedData.perform (dart:async/stream_impl.dart:515:14)
E/flutter (12668): #9      _PendingEvents.handleNext (dart:async/stream_impl.dart:620:11)
E/flutter (12668): #10     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:591:7)
E/flutter (12668): #11     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
E/flutter (12668): #12     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
E/flutter (12668):
```

Additional system variables:
Same as https://github.com/couchbase-examples/flutter_cbl_learning_path/issues/1

**Comment by biozal** [comment]
@jdar This should now be fixed.  Thanks again @blaugold for the help on this issue!

**Comment by blaugold** [comment]
I think this is an unrelated issue. I can still replicate it.

**Comment by biozal** [comment]
OK let me take a look at this.  I think this is because the user profile code that doesn't use the modal is still not in the main branch.

**Comment by blaugold** [comment]
I have a fix in #5. The problem is calling `Navigator.of(context).pop()` from a build function. We just need to use `BlocListener` instead of a `BlocBuilder`.

**Comment by jdar** [comment]
I confirm that the fix in #5 works for me. Tested on Android. Please do merge!
